docs/source/_notebooks/keras-text-classifiers.rst

```diff
@@ -0,0 +1,3685 @@
+
+Explaining Keras text classifier predictions with Grad-CAM
+==========================================================
+
+We can use ELI5 to explain text-based classifiers, i.e. models that take
+in a text and assign it to some class. Common examples include sentiment
+classification, labelling into categories, etc.
+
+The underlying method used is 'Grad-CAM'
+(https://arxiv.org/abs/1610.02391). This technique shows what parts of
+the input are the most important to the predicted result, by overlaying
+a "heatmap" over the original input.
+
+See also the tutorial for images
+(https://eli5.readthedocs.io/en/latest/tutorials/keras-image-classifiers.html).
+Certain sections such as 'removing softmax' and 'comparing different
+models' are relevant for text as well.
+
+Set up
+------
+
+First some imports
+
+.. code:: ipython3
+
+    import os
+    
+    import numpy as np
+    import pandas as pd
+    from IPython.display import display, HTML  # our explanations will be formatted in HTML
+    
+    # you may want to keep logging enabled when doing your own work
+    import logging
+    import tensorflow as tf
+    tf.get_logger().setLevel(logging.ERROR) # disable Tensorflow warnings for this tutorial
+    import warnings
+    warnings.simplefilter("ignore") # disable Keras warnings for this tutorial
+    import keras
+    from keras.preprocessing.sequence import pad_sequences
+    
+    import eli5
+
+
+.. parsed-literal::
+
+    Using TensorFlow backend.
+
+
+The rest of what we need in this tutorial is stored in the
+``tests/estimators`` package, whose source you can check for your own
+reference. You may need extra steps here to load your custom model and
+data.
+
+.. code:: ipython3
+
+    # we need to go back to top level in order to import some local ELI5 modules
+    
+    old = os.getcwd()
+    os.chdir('..')
```

Comment:
This is for modules defined in tests, right? Then a more common way to import them would be to change the pythonpath (via ``sys.path``).